src/main/java/com/github/fge/jsonpatch/JsonPatchDeserializer.java

```diff
@@ -0,0 +1,36 @@
+package com.github.fge.jsonpatch;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import java.io.IOException;
+
+/**
+ * JsonPatchDeserializer allows you to register a custom JsonPatchFactory for jackson to use
+ * during deserialization.
+ */
+public class JsonPatchDeserializer
+    extends JsonDeserializer<JsonPatch>
+{
+    private JsonPatchFactory factory;
+    public JsonPatchDeserializer(JsonPatchFactory factory)
+    {
+        this.factory = factory;
+    }
+
+    @Override
+    public JsonPatch deserialize(final JsonParser jp,
+        final DeserializationContext ctxt)
+        throws IOException, JsonProcessingException
+    {
+        final JsonNode node = jp.readValueAsTree();
```

Comment:
I'm wondering if you tried to move the factory.fromJson() from taking a JsonNode to taking a JsonParser if you could avoid a bunch of overhead of deserializing into a DOM.